src/provider/file/mod.rs

```diff
@@ -599,19 +622,41 @@ mod test {
         );
     }
 
+    fn mock_secret_key() -> SecretKeyEntry {
+        SecretKeyEntry::new(
+            "Bogo Key".to_owned(),
+            vec![
+                SignatureRole::Host,
+                SignatureRole::Proxy,
+                SignatureRole::Creator,
+            ],
+        )
+    }
+
     #[tokio::test]
     async fn test_should_create_yank_invoice() {
         // Create a temporary directory
         let root = tempdir().unwrap();
-        let scaffold = testing::Scaffold::load("valid_v1").await;
+        let mut scaffold = testing::Scaffold::load("valid_v1").await;
         let store = FileProvider::new(
             root.path().to_owned(),
             crate::search::StrictEngine::default(),
+            KeyRing::default(),
         )
         .await;
         let inv_name = scaffold.invoice.canonical_name();
+
+        let sk = mock_secret_key();
         // Create an file
-        let missing = store.create_invoice(&scaffold.invoice).await.unwrap();
+        let missing = store
+            .create_invoice(
+                &mut scaffold.invoice,
```

Comment:
Ok, so this made me realize that this pattern of `&mut invoice` is really odd. Basically we end up mutating the invoice the user gives us rather than having them depend on the returned one for getting the signed invoice. Although I don't think it has to happen here, we should probably change this to be `mut invoice:Invoice` and just have it take ownership. I can do that in a general "cloning cleanup" I'd like to do over the whole repo anyway once this is merged

Reply:
Okay. I will leave it to you to change it however you want it.